src/ClientInterface.php

```diff
@@ -11,9 +11,8 @@
 
 namespace Sylius\Api;
 
-use GuzzleHttp\Message\ResponseInterface;
-use GuzzleHttp\Url;
-use GuzzleHttp\Event\SubscriberInterface;
+use GuzzleHttp\Psr7\Uri;
```

Comment:
Instead of typehintnig with URI implementation from Guzzle library, I would suggest to use an [UriInterface](https://github.com/php-fig/http-message/blob/master/src/UriInterface.php) from `Psr\Http\Message` namespace.